### ISSUE TITLE
Sqlite adapter doesn't seem to like in-memory databases

### ISSUE THREAD

**Issue by igorsantos07** [Codeception/Codeception#3319]
According to [PHP's doc](http://php.net/manual/en/ref.pdo-sqlite.connection.php), we can use a simpler DSN for Sqlite in-memory database (`sqlite::memory:`). That seems neat for a short-living test, thus I gave it a shot.

Actually, it does work. But instead of using an actual memory memory connection, it created a `:memory:` file in my project root folder haha

I tested using that DSN from the CLI and it worked as expected, leaving no trace on the current folder.

Codeception 2.2.3, PHPUnit 5.4.6, PHP 5.6.23

**Comment by Naktibalda** [comment]
Can in-memory database be shared by Codeception and your application?

**Comment by DavertMik** [comment]
Don't use memory database for testing.

1) You can't debug it 
2) SQLite is probably differs from you primary application database so tests may be not relevant 
3) it will work for integration and functional tests only, but I think that the same way you can just set an ORM with nested transaction like we do for Doctrine, Eloquent, etc. 

And yes, we don't support SQLite memory.

**Comment by igorsantos07** [comment]
That seems more like a "we implemented a bug to disallow SQLite in-memory" instead of "we don't support it".
If that's the case, please catch the configuration and issue a proper warning, instead of allowing us to implement it and keep wondering why the hell it's creating a weirdly-named file instead of following the PDO behaviour.

**Comment by Naktibalda** [comment]
If you could implement the change and test it against a real application, I
am sure that we would be happy to accept a pull request.

**Comment by igorsantos07** [comment]
Thanks! Will do. I'm using SQLite currently to test a small Phalcon package that interacts with a new table, so it fits nicely. As soon as I have some tests passing I'll fix the SQLite implementation and send a PR. Could this issue be reopened, @Naktibalda?

**Comment by Naktibalda** [comment]
@igorsantos07 any luck?
I can make it throw exception if user tries to set the file to ':memory:'.

**Comment by igorsantos07** [comment]
I ended up leaving this for later and forgot it, @Naktibalda. Sorry!

So, I implemented it but I stumbled upon another issue (I saw this mentioned somewhere else...). Memory databases would only work if the connection was shared between Codeception and the actual  tested codebase... And that doesn't seem feasible.

What does it mean?
Even if Codeception allows `sqlite::memory:`, and the tested codebase uses the same DSN, both will have different memory spaces and won't be talking with the same database.
Given that, I _think_ an exception should be thrown if the DSN is `sqlite::memory:`, or a warning should be given on the output at least - so we wouldn't forbid usage, but warn this is probably not going to work.


To register, this is what I wrote:
```php
//Codeception/Lib/Driver/Sqlite.php
    public function __construct($dsn, $user, $password)
    {
        $this->filename = substr($dsn, 7);
        if ($this->filename != ':memory:') { //special handling for in-memory databases
            $this->filename = Configuration::projectDir() . $this->filename;
        }
        $this->dsn = 'sqlite:' . $this->filename;
        parent::__construct($this->dsn, $user, $password);
    }
```
